apps/api/src/function/enqueuer/src/rabbitmq.ts

```diff
@@ -16,59 +16,87 @@ export class RabbitMQEnqueuer extends Enqueuer<RabbitMQOptions> {
       "Designed for reliable background job processing and asynchronous event consumption."
   };
 
-  private connections = new Set<amqp.ChannelModel>();
-  private channels = new Set<amqp.Channel>();
+  private subscriptions: Subscription[] = [];
 
   constructor(
     private queue: EventQueue,
     private rabbitmqQueue: RabbitMQQueue,
     private schedulerUnsubscription: (targetId: string) => void
   ) {
     super();
+
+    setInterval(() => {
+      this.subscriptions.forEach(s => {
+        if (!s.closed) return;
+
+        this.unsubscribe(s.target);
+        this.subscribe(s.target, s.options);
+      });
+    }, 60_000);
   }
 
   async subscribe(target: event.Target, options: RabbitMQOptions) {
+    const subscription: Subscription = {target, options, closed: false};
+    this.subscriptions.push(subscription);
+
     const connection = await amqp.connect(options.url).catch(err => {
+      subscription.closed = true;
+
       this.onErrorHandler(target, "Connection failed. " + err.message);
       return null;
     });
+
     if (!connection) return;
 
+    subscription.connection = connection;
+
     connection.on("close", () => {
-      if (this.connections.has(connection)) {
-        this.onErrorHandler(target, "The connection was closed unexpectedly.");
-      }
+      this.setAsClosed({
+        connection,
+        target,
+        errorMessage: "The connection was closed unexpectedly."
+      });
     });
 
-    Object.defineProperty(connection, "target", {writable: false, value: target});
-    this.connections.add(connection);
-
     const channel = await connection.createChannel().catch(err => {
-      this.onErrorHandler(target, "Channel creation failed. " + err.message);
+      this.setAsClosed({
+        connection,
+        target,
+        errorMessage: "Channel creation failed. " + err.message
+      });
       return null;
     });
     if (!channel) return;
 
+    subscription.channel = channel;
```

Comment:
Could you move this line to the end of the function, where we complete configuring the channel?